test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -387,6 +388,13 @@ func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
 				mu.Unlock()
 			}
 
+			if !concreteOp.SkipWaitToCompletion {
+				// SkipWaitToCompletion=false indicates this step have waited for the Pods to be scheduled.
+				// So we reset the metrics in global registry; otherwise metrics gathered in this step/test
+				// will be rolled over to next step/test.
```

Comment:
doing this even when this step doesn't collect metrics is a bit wasteful, no? I think we could change the condition to `CollectMetrics && !SkipWaitToCompletion`? I don't mind leaving this as it is to be more defensive, I'm only interested in knowing if what I said is true as a sanity check

Reply:
Not quite, in some case, it's intentional to only initialize Pods, but not collect metrics of them:

```yaml
- name: SchedulingBasic
  workloadTemplate:
  - opcode: createNodes
    countParam: $initNodes
  - opcode: createPods
    countParam: $initPods
    podTemplatePath: config/pod-default.yaml
```

This step generates metrics, so we should get them cleared.